zkwrapper/source/zk_queue.cc

```diff
@@ -41,22 +42,33 @@ bool pop(const std::shared_ptr <ZKWrapper> &zk, const std::string &q_path, std::
 		LOG(ERROR) << "Failed to get children of queue node: " << q_path;
 		return false;
 	}
+
+	ZKLock q_lock(*zk, q_path);
+	q_lock.lock();
+
 	if (children.size() < 1) {
-		std::cerr << "ERROR: Queue is empty, nothing to pop!" << std::endl;
+		LOG(ERROR) << "Queue is empty, nothing to pop!";
 		// TODO: set a custom error code?
 		return false;
 	}
 
 	std::string peeked_path = q_path + "/" + children.back();
```

Comment:
Why children.back()? Don't we need to sort children and then use children.front()?